text/XXXX-storage_command.md

```diff
@@ -0,0 +1,178 @@
+```
+$ qri storage purge --help
+Purge deletes dataset version data, freeing up disk space. Purge does not edit 
+history. To edit version history, use the remove command.
+
+Dataset versions take up disk space, and at some point need to be managed.
+Becase datasets and storage capacity vary dramatically the need for this
+
+You can also use purge to request a remote drop version data. This is often
+
+Usage:
+  qri purge [REMOTE] DATASET [flags]
+
+Examples:
+  # drop
+
+Flags:
+  -a, --all               purge all versions, overrides revisions
+  -r, --revisions         purge a number of versions from the latest commit
+  -t, --tail              purge stored versions starting from the earliest
+
+```
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+The purpose of stats are to answer a few primary questions. All of them deal with a _set of blocks_. 
+
+
+The foundation for all of this work should be a `dag.Info` & `dag.Manifest`. Info provides the block size information, and manifests provide the . `dag.Info` currently looks like this:
+
+```go
+// Info is os.FileInfo for dags: a struct that describes important
+// details about a graph. Info builds on a manifest.
+//
+// when being sent over the network, the contents of Info should be considered gossip,
+// as Info's are *not* deterministic. This has important implications
+// Info should contain application-specific info about a datset
+type Info struct {
+	// Info is built upon a manifest
+	Manifest *Manifest      `json:"manifest"`
+	Labels   map[string]int `json:"labels,omitempty"` // sections are lists of logical sub-DAGs by positions in the nodes list
+	Sizes    []uint64       `json:"sizes,omitempty"`  // sizes of nodes in bytes
+}
+```
+
+Here's the current Manifest data structure:
+```go
+type Manifest struct {
+  Links [][2]int `json:"links"` // links between nodes
+  Nodes []string `json:"nodes"` // list if CIDS contained in the DAG
+}
+```
+
+Manifests assume they describe one graph with a single root. To represent multiple graphs in the same manifest, we'll need to make one change:
+
+```go
+type Manifest struct {
+  Graphs [][][2]int `json:"links"` // links between nodes
+  Nodes []string `json:"nodes"` // list if CIDS contained in the DAG
+}
+```
+
+Calculating storage states
+1. Load logbook info for a dataset
+2. Build a `dag.Manifest` of all stored roots, keeping a slice of block sizes while you're at it
+3. Turn that into a `dag.Info`
+4. Call methods on `dag.Info` that produce storage stats
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Why should we *not* do this?
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+<!-- - Why is this design the best in the space of possible designs?
+- What other designs have been considered and what is the rationale for not choosing them?
+- What is the impact of not doing this? -->
+
+# Prior art
+[prior-art]: #prior-art
+
+
```

Comment:
AFAIK Docker has a similar "process" to clean up old/unused/build images from your machine. The ramifications are not really comparable but offer some parallel. One of the things they do is clean out "inactive/stopped" images which might make sense that we start from some sort of least used datasets when thinking about purging and maybe having different levels of how aggresive the purge is.

Reply:
+1, I end up using `docker system prune` a lot, it would be good to have some easily choosable options for clearing older versions of datasets, or datasets that have not been accessed recently, etc